robusta_krr/core/models/result.py

```diff
@@ -17,8 +17,8 @@ class Recommendation(pd.BaseModel):
 
 
 class ResourceRecommendation(pd.BaseModel):
-    requests: dict[ResourceType, RecommendationValue]
-    limits: dict[ResourceType, RecommendationValue]
+    requests: dict[ResourceType, Union[RecommendationValue, Recommendation]]
```

Comment:
This is kind of related, it is issue with the following code which breaks in test since we setting field to other type but there isn't validation on non init.
```
                #TODO: consider... changing field after model created doesn't validate it.
                getattr(recommendation_processed, selector)[resource_type] = Recommendation(
                    value=recommended, severity=current_severity
                )
```